### PR TITLE
fix(eth-multisig-v4): modify deploy script for v4 contracts

### PR DESCRIPTION
**Pull request by gianchandania**
WIN-1140

This PR modifies the deploy script to support v4 contracts. It also adds a new private key that will be used to deploy the v4 contracts

### CODE REVIEW

File: .github/workflows/deploy_and_release.yml [hunk 1]
```diff
@@ -61,8 +65,12 @@ jobs:
         env:
           MAINNET_PRIVATE_KEY_FOR_CONTRACT_DEPLOYMENT: ${{ secrets.MAINNET_PRIVATE_KEY_FOR_CONTRACT_DEPLOYMENT }}
           TESTNET_PRIVATE_KEY_FOR_CONTRACT_DEPLOYMENT: ${{ secrets.TESTNET_PRIVATE_KEY_FOR_CONTRACT_DEPLOYMENT }}
+          PRIVATE_KEY_FOR_V4_CONTRACT_DEPLOYMENT: ${{ secrets.PRIVATE_KEY_FOR_V4_CONTRACT_DEPLOYMENT }}
           QUICKNODE_ETH_MAINNET_API_KEY: ${{ secrets.QUICKNODE_ETH_MAINNET_API_KEY }}
           QUICKNODE_ETH_HOLESKY_API_KEY: ${{ secrets.QUICKNODE_ETH_HOLESKY_API_KEY }}
+          QUICKNODE_ARBITRUM_SEPOLIA_API_KEY: ${{ secrets.QUICKNODE_ARBITRUM_SEPOLIA_API_KEY }}
+          QUICKNODE_OPTIMISM_SEPOLIA_API_KEY: ${{ secrets.QUICKNODE_OPTIMISM_SEPOLIA_API_KEY }}
+          QUICKNODE_ZKSYNC_SEPOLIA_API_KEY: ${{ secrets.QUICKNODE_ZKSYNC_SEPOLIA_API_KEY }}
```
Comment (sachushaji) [inline .github/workflows/deploy_and_release.yml:73]: Keys are same for mainnet and sepolia?

File: .github/workflows/deploy_and_release.yml [hunk 2]
```diff
@@ -115,8 +123,12 @@ jobs:
...
```
Comment (sachushaji) [inline .github/workflows/deploy_and_release.yml:131]: This one is for mainnet right? Should be mainnet key?

File: scripts/deploy.ts [hunk 3]
```diff
@@ -76,17 +76,19 @@ async function main() {
   output.walletFactory = walletFactory.address;
   console.log('WalletFactory deployed at ' + walletFactory.address);
 
-  const Forwarder = await ethers.getContractFactory('Forwarder');
+  const Forwarder = await ethers.getContractFactory('ForwarderV4');
```
Comment (sachushaji) [inline scripts/deploy.ts:79]: Is this a deploy function which common for everything?, If we have to deploy a forwarder of earlier versions, say for eg: if any testnet deprecates where we used a previous version, we may need to chnage the code again right?, can we pass in the forwarder as a parameter

File: .github/workflows/deploy_and_release.yml (hunk 1 above)
Comment (gianchandania) [reply .github/workflows/deploy_and_release.yml:73]: Ohh they are different, my bad. Will correct this

File: .github/workflows/deploy_and_release.yml (hunk 2 above)
Comment (gianchandania) [reply .github/workflows/deploy_and_release.yml:131]: Yeah, should be mainnet key. Correcting this

File: scripts/deploy.ts (hunk 3 above)
Comment (gianchandania) [reply scripts/deploy.ts:79]: I mainly thought that going forward we are going to deploy new forwarder only, but the point makes sense, will address this.

File: .github/workflows/deploy_and_release.yml (hunk 1 above)
Comment (gianchandania) [reply .github/workflows/deploy_and_release.yml:72]: Currently we don't have mainnet keys, will get those once we are done deploying on sepolia

File: scripts/deploy.ts (hunk 3 above)
Comment (gianchandania) [reply scripts/deploy.ts:85]: Have added comment for now as this will need changes even in github workflow as we can take forwarder version from a label

**Review by sachushaji — APPROVED** [review]

**Pull request merged**